map.go

```diff
@@ -717,13 +755,34 @@ func (mi *MapIterator) Next(keyOut, valueOut interface{}) bool {
 		return false
 	}
 
+	// gessing a random start key
+	if !supportNullKey.value && mi.randomStartGuess {
```

Comment:
Can we check `mi.prevKey == nil` rather than adding the `randomStartGuess` field?